### PR TITLE
[APB-2060][HO] added ability to CRUD properties in mongo

### PR DESCRIPTION
**Pull request by habeeb01**

### CODE REVIEW

File: app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+class PropertiesController @Inject()(repository: PropertiesRepository) extends BaseController {
+
+
+  def createProperty = Action.async(parse.json) { implicit request =>
+    withJsonBody[Property] { property =>
+      repository.findProperty(property.key).flatMap { op =>
+        if (op.isEmpty) {
+          repository.createProperty(property).map(_ => Created)
+        } else {
+          Future successful BadRequest(Json.toJson(ErrorBody("PROPERTY_EXISTS", "Property already exists")))
```
Comment (arturopala) [inline app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala]: maybe use 409 Conflict here?

File: app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+      }
+    }
+  }
+
+  def updateProperty(key: String) = Action.async(parse.json) { implicit request =>
```
Comment (arturopala) [inline app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala]: do we need separated create/update semantics? maybe UPSERT would be more convenient?

File: app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala (hunk 1 above)
Comment (habeeb01) [reply app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala]: I have now applied this change

File: app/uk/gov/hmrc/agentassurance/repositories/PropertiesRepository.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import play.api.libs.json.Format
+import play.api.libs.json.Json.format
+import play.modules.reactivemongo.ReactiveMongoComponent
+import uk.gov.hmrc.mongo.{AtomicUpdate, ReactiveRepository}
+import uk.gov.hmrc.agentassurance.model._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PropertiesRepository @Inject()(mongoComponent: ReactiveMongoComponent)
```
Comment (arturopala) [inline app/uk/gov/hmrc/agentassurance/repositories/PropertiesRepository.scala]: index by key would be welcome, we can also add `StrictlyEnsuredIndexes` trait from template

File: conf/app.routes [hunk 4]
```diff
@@ -8,4 +8,10 @@ GET        /activeCesaRelationship/nino/:nino/saAgentReference/:saAgentReference
 
 GET        /acceptableNumberOfClients/service/IR-PAYE                                   @uk.gov.hmrc.agentassurance.controllers.AgentAssuranceController.acceptableNumberOfPAYEClients
 
-GET        /acceptableNumberOfClients/service/IR-SA                                     @uk.gov.hmrc.agentassurance.controllers.AgentAssuranceController.acceptableNumberOfIRSAClients
\ No newline at end of file
+GET        /acceptableNumberOfClients/service/IR-SA                                     @uk.gov.hmrc.agentassurance.controllers.AgentAssuranceController.acceptableNumberOfIRSAClients
+
+
+GET        /properties/:key                                                             @uk.gov.hmrc.agentassurance.controllers.PropertiesController.getProperty(key: String)
```
Comment (arturopala) [inline conf/app.routes]: what about making it more specific, not generic, we will deal with two separate concepts: refusal2dealWith and manually assured

Comment (habeeb01) [reply conf/app.routes]: I was struggling to come up with a good url (I mentioned it at the stand up). I will come up with something better

Comment (arturopala) [reply conf/app.routes]: I would imagine here 2 sets of endpoints, one for R2DWL, another one for manually assured, WDYT?

File: app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala (hunk 2 above)
Comment (habeeb01) [reply app/uk/gov/hmrc/agentassurance/controllers/PropertiesController.scala]: having thought about it, using upsert would mean that we will not be able to explicitly tell when someone is trying to update a property. Maybe its best to keep the 4 endpoints

File: conf/app.routes (hunk 4 above)
Comment (habeeb01) [reply conf/app.routes]: 👌 and both sets of endpoints will do the exact same thing. We can go with this

**Pull request merged**